forcepho/dispatcher.py

```diff
@@ -568,183 +588,330 @@ def __init__(self, comm, n_children):
+    # but isn't the point that we don't have to pass around a logger object?
+    global logger
+    logger = logging.getLogger('dispatcher-parent')
+    logger.info(f'Starting parent on {socket.gethostname()}')
+    
+    rank = comm.Get_rank()
+    n_child = comm.Get_size() - 1
+    patchcat = {}
+
+    # Make Queue
+    queue = MPIQueue(comm, n_child)
+    
+    ###################
+    # steal a real patch from the demo
+    import os
+    os.chdir('../demo')
+    import sys
+    sys.path.insert(0,'.')
+    from utils import rectify_catalog
+    from config_test import config
+
+    # --- Get the patch dispatcher ---  (parent)
+    cat, bands, chdr = rectify_catalog(config.raw_catalog)
+    sceneDB = SuperScene(sourcecat=cat, bands=bands,
+                         maxactive_per_patch=config.maxactive_per_patch,
+                         maxradius=config.patch_maxradius,
+                         target_niter=config.sampling_draws,
+                         statefile="superscene.fits",
+                         bounds_kwargs={},
+                         maxactive_fraction=0.5)
+    ##########################
+    
+
+    checkout_time = 0
+    
+    # Do it in context so failure still writes current superscene
+    with sceneDB:
+        logger.debug(f'SuperScene has {sceneDB.n_sources} sources')
+        
+        # TODO: LHG changed this from patchid to taskid, to make it clear it's more of a counter than an index of a patch in the scene
+        # but this could/should be changed back if that's not the right interpretation
+        taskid = 0
+        
+        # EVENT LOOP
+        while sceneDB.undone:
+            # Generate one patch proposal
+            _tstart = timer()
+            ntry = 1000  # how many tries before we decide there are no regions to be checked out?
+            for _ in range(ntry):
+                region, active, fixed = sceneDB.checkout_region()
+                mass = None  # TODO: this should be returned by the superscene
+                if active is not None:
+                    break
+            else:
+                logger.debug(f'Failed to checkout region')
+            checkout_time += timer() - _tstart
+            
+            # construct the task
+            if active is not None:
+                if not sceneDB.sparse:
+                    logger.debug(f'Scene no longer sparse with {queue.n_children - len(queue.idle) + 1} checkout(s)')
+                logger.info(f'Checked out region {len(active)} actives')
+                taskid += 1
+                
+                bounds = sceneDB.bounds_catalog[active["source_index"]]
+                lower, upper, cov = sceneDB.bounds_and_covs(active["source_index"],
+                                                    bands=bands,  # TODO: can we do this without patcher.bandlist?
+                                                    ref=np.zeros(2))  # TODO: can we do this without patcher.patch_reference_coordinates?
```

Comment:
Super unsure about these. Trying to keep the SuperScene on the server, and the JadesPatch on the client.